### PR TITLE
search feature

### PR DESCRIPTION
**Pull request by T-Y89**
I thought this application needed a label search feature. So I made it.

### CODE REVIEW

**Review by nk-ty — COMMENTED** [review]
Thanks for your good suggestion!

File: app/controllers/test_reports_controller.rb [hunk 1]
```diff
@@ -61,6 +66,17 @@ def child_loop(job_tree, job_id, indent_num)
...
 
+  def ganerate_pagenation(search_word = nil)
+    if search_word.nil?
+      Kaminari.paginate_array(Job.all_root_jobs.to_a, total_count: Job.all_root_jobs.length).page(params[:page]).per(PER_PAGE)
+    else
+      Kaminari.paginate_array(
+        Job.all_root_jobs.searched_root_jobs(params[:search_word]).to_a,
+        total_count: Job.all_root_jobs.searched_root_jobs(params[:search_word]).length
+      ).page(params[:page]).per(PER_PAGE)
+    end
```
Comment (nk-ty) [inline app/controllers/test_reports_controller.rb:77]: I think there's a little bit of duplicate parts.

How about fixing it like this:
```ruby
          elements = if search_word.nil?
                        Job.all_root_jobs.to_a
                      else
                        Job.all_root_jobs.searched_root_jobs(params[:search_word]).to_a
                      end
          Kaminari.paginate_array(elements, total_count: elements.length).page(params[:page]).per(PER_PAGE)
```

File: app/controllers/test_reports_controller.rb [hunk 2]
```diff
@@ -2,16 +2,21 @@
...
 class TestReportsController < ApplicationController
   before_action :check_round, only: %i[show update]
+  PER_PAGE = 30
   def index
-    per_page = 30
-    @page = Kaminari.paginate_array(Job.all_root_jobs.to_a, total_count: Job.all_root_jobs.length).page(params[:page]).per(per_page)
-    start_num = params[:page].nil? || params[:page] == 1 ? 0 : per_page * (params[:page].to_i - 1)
-    root_jobs = Job.root_jobs(start_num, per_page)
+    start_num = params[:page].nil? || params[:page] == 1 ? 0 : PER_PAGE * (params[:page].to_i - 1)
+    root_jobs, @page = if params[:search_word]
```
Comment (nk-ty) [inline app/controllers/test_reports_controller.rb:8]: If there is no `search_word` parameter,  `params[:search_word]` is nil?
If so, how about implementing it like this:

```ruby
          root_jobs if params[:search_word]
                      Job.get_searched_root_jobs(start_num, PER_PAGE, params[:search_word])
                    else
                      Job.root_jobs(start_num, PER_PAGE),
                    end

          @page = ganerate_pagenation(params[:search_word])
```

I also think you can remove the default parameters of `ganerate_pagenation` as well.

File: app/models/job.rb [hunk 3]
```diff
@@ -31,18 +31,27 @@ class Job < ApplicationRecord
...
     Job.all.where("command_and_option not like '%rerun%'").order('jobs.id DESC')
   end
 
-  def self.all_children_jobs(start, limit)
-    Job.all.where("command_and_option like '%rerun%'").where(id: start..).order('jobs.id ASC').limit(limit)
+  def self.searched_root_jobs(search_word)
+    all_root_jobs.where('command_and_option LIKE ?', "%#{search_word}%")
+  end
+
+  def self.get_searched_root_jobs(start_num, per_page, search_word)
```
Comment (nk-ty) [inline app/models/job.rb:42]: I thought there was no difference between "searched_root_jobs" and "get_searched_root_jobs".
How about fixing it like this:
```suggestion
  def self.searched_root_jobs_in_page(start_num, per_page, search_word)
```

**Review by nk-ty — COMMENTED** [review]
Thanks for you fix!
Could you please check my comment?

File: app/controllers/test_reports_controller.rb (hunk 1 above)
Comment (T-Y89) [reply app/controllers/test_reports_controller.rb:77]: i agree your suggestion

File: app/controllers/test_reports_controller.rb [hunk 4]
```diff
@@ -67,14 +66,12 @@ def child_loop(job_tree, job_id, indent_num)
   end
 
   def ganerate_pagenation(search_word = nil)
```
Comment (nk-ty) [inline app/controllers/test_reports_controller.rb:68]:
```suggestion
  def ganerate_pagenation(search_word)
```

**Review by nk-ty — CHANGES_REQUESTED** [review]
Thanks for you fix!

File: app/controllers/test_reports_controller.rb (hunk 4 above)
Comment (nk-ty) [reply app/controllers/test_reports_controller.rb:68]: You fixed so the default argument is no longer needed.

**Review by nk-ty — APPROVED** [review]
LGTM

**Pull request merged**